src/compiler/proposals-and-stabilization/custom-test-jobs.md

```diff
@@ -0,0 +1,88 @@
+- otherwise do not have well documented test job maintainers and failure protocol;
+
+can introduce a lot of friction and frustration for other contributors who now will have to deal with the failure or understand why the test job failed in the first place, which may now be blocking their PR. Everyone utilizing the [rust-lang/rust] CI is expected to use it responsibly.
+
+To help with this, please follow the process described below if you'd like to propose adding a custom test job/component to the [rust-lang/rust] CI, and copy and fill out the Custom Test Job/Component Policy.
+
+## Process for adding a custom test job to rust-lang/rust CI
+
+![Flowchart of the process for adding a custom test job/component to rust-lang/rust CI](./custom-test-jobs/test-job-flow.png)
+
+## Custom Test Job/Component Template
+
+*Please copy and fill out this template as part of the MCP.*
+
+The proposed custom test job or custom test component for the [rust-lang/rust] CI must:
+
+- Be approved by the compiler team through a proposed Major Change Proposal (MCP), where the MCP is seconded by a compiler team member, and the MCP being accepted with no blocking concerns.
+- Have no blocking concerns from the library team by creating an issue on [rust-lang/rust] proposing to add the custom test job/component to [rust-lang/rust] CI linking to the MCP, and nominating the issue for review by the library team via `@rustbot label +I-libs-nominated`.
+- Have the implementation PR be reviewed and approved by the infrastructure team, linking to the MCP and issue.
+- Be properly documented on [rustc-dev-guide] (preferably as part of the implementation PR), linking to the MCP and issue.
+
+### Test job/component rationale
+
+- What does this custom test job/component do?
+- What [rust-lang/rust] changes can potentially break the custom test job/component? E.g. changes to rustc, standard library, bootstrap or tools (like clippy/rustfmt/cargo).
+- Why does this custom test job/component need to be part of the [rust-lang/rust] PR and/or Full Merge CI?
+- If the custom test job/component will block on failure, why does it need to block?
+- If the custom test job/component will not block on failure initially but is intended to eventually become blocking, why so? And when will it become blocking?
+
+### Test job/component maintainers
+
+The proposed custom test job/component for [rust-lang/rust] CI must have at least one dedicated maintainer. The test job/component maintainers understand that they will be pinged or otherwise contacted about the custom test job/component, particularly for (but not limited to) its failures. Please list who will be maintaining this test job/component here.
+
+**We, the custom test job/component maintainers, acknowledge and consent that we will be pinged about the custom test job/component failures or be consulted about it in general.**
```

Comment:
Discussion: I haven't carried over the test job/component ping group being **mandatory** over to this draft yet. I can modify the draft to require a ping group if people find that is necessary.

Reply:
I think a mandatory ping group is good.

Reply:
It turns out that only team members and certain WG-* members can even use `@rustbot ping xxx`, which IMO makes mandatory ping groups mostly useless. Non-team/wg-members can't ping `@rust-lang/xxx` nor `@rustbot ping xxx` anyway, while org members can directly `@rust-lang/xxx` or just `@` the test job maintainers directly.

Reply:
I removed mentions of ping groups in (round 2).